app/lib/admin/backend.dart

```diff
@@ -0,0 +1,200 @@
+        if (lastId != null) {
+          query.filter('__key__ >', key);
+        }
+      }
+
+      final users = await query.run().toList();
+      // We may return a page with users less then a limit, but we always
+      // set the continuation token to the correct value.
+      final newContinuationToken = users.length < limit
+          ? null
+          : _continuationCodec.encode(users.last.userId);
+      users.removeWhere((u) => u.isDeleted);
+
+      return api.AdminListUsersResponse(
+        users: users
+            .map(
+              (u) => api.AdminUserEntry(
+                userId: u.userId,
+                email: u.email,
+                oauthUserId: u.oauthUserId,
+              ),
+            )
+            .toList(),
+        continuationToken: newContinuationToken,
+      );
+    });
+  }
+
+  /// Removes user from the Datastore and updates the packages and other
+  /// entities they may have controlled.
+  Future removeUser(String userId) async {
+    await _withAdmin((u) async {
+      // TODO: remove once [withAuthenticatedUser] is using [User]
+      final user = await accountBackend.lookupUserById(userId);
+      if (user == null) return;
+      if (user.isDeleted) return;
+
+      _logger.info(
+          '${u.userId} (${u.email}) initiated the delete of ${user.userId} (${user.email})');
+
+      // Package.uploaders
+      final pkgQuery = _db.query<Package>()..filter('uploaders =', user.userId);
+      await for (final p in pkgQuery.run()) {
+        await _removeUploaderFromPackage(p.key, user.userId);
+      }
+
+      // PublisherMember
+      // Publisher.contactEmail
+      final memberQuery = _db.query<PublisherMember>()
+        ..filter('userId =', user.userId);
+      await for (final m in memberQuery.run()) {
+        await _removeMember(user, m);
+      }
+
+      // OAuthUserID
+      if (user.oauthUserId != null) {
+        final key = _db.emptyKey.append(OAuthUserID, id: user.oauthUserId);
+        final mapping = (await _db.lookup<OAuthUserID>([key])).single;
+        if (mapping != null) {
+          await _db.commit(deletes: [mapping.key]);
+        }
+      }
+
+      // User
+      await _markUserDeleted(user);
+    });
+  }
+
+  Future _removeUploaderFromPackage(Key pkgKey, String userId) async {
+    await _db.withTransaction((tx) async {
+      final p = (await tx.lookup<Package>([pkgKey])).single;
+      p.removeUploader(userId);
+      if (p.uploaders.isEmpty) {
+        p.isDiscontinued = true;
+      }
+      tx.queueMutations(inserts: [p]);
+      await tx.commit();
+    });
+  }
+
+  Future _removeMember(User user, PublisherMember member) async {
+    final seniorMember =
+        await _remainingSeniorMember(member.publisherKey, member.userId);
+    await _db.withTransaction((tx) async {
+      final p = (await tx.lookup<Publisher>([member.publisherKey])).single;
+      if (seniorMember == null) {
+        p.contactEmail = null;
```

Comment:
Okay, if we have this (and I think you're right, there is no other option), then we must document in `Publisher.contactEmail` that the property can be `null`. And we'll likely have to handle that case in templates too..